spec/rest-read-resource.md

```diff
@@ -0,0 +1,41 @@
+
+For example, suppose `/alice/notes/` is a container. A `GET /alice/notes/` might return an HTML page with a list of links to each item (`shoppinglist.txt`, etc.). If the client specifically requests JSON, the response might look like an array or object enumerating the children.
+
+In all cases, the server **SHOULD** include relevant metadata in the response headers. For a container, an `ETag` can represent a version of the listing (which could change when any member is added or removed), and perhaps a header indicating it is a container.
+
+**HEAD (any resource or container)** – *Headers/metadata only:*  
+ An HTTP HEAD request is identical to GET except that the server **MUST NOT** send a message body in the response. It’s used to fetch the headers (metadata) for a resource or to check if it exists, without downloading the content. The LWS server should support HEAD for both containers and non-containers. For a non-container resource, HEAD returns the same headers that a GET would (Content-Type, Content-Length, `ETag`, etc.) but no body. This is useful for, e.g., checking if a cached copy is still valid (via ETag or Last-Modified) or simply checking existence (a HEAD yielding `200 OK` means the resource exists and the client is allowed to see it; a 404 means it doesn’t exist or not allowed). For a container, HEAD might return headers like an ETag for the container listing, or simply a `200 OK` with no body, which at least confirms the container exists and the client has access.
+
+If a resource doesn’t exist, a HEAD request should return `404 Not Found` (just like GET would, but again with no body). If the client isn’t authorized, the server should return 403 or 404 as appropriate (mirroring the GET behavior, where `403 Forbidden` is for known-but-forbidden, and 404 Not Found might be used to avoid revealing existence to unknown clients).
```

Comment:
Are HTTP Semantics not defined in a way that the HEAD response is "derived" from the definition of GET?